src/ensembl/src/content/app/entity-viewer/shared/components/entity-viewer-sidebar/entity-viewer-sidebar-modal/modal-views/EntityViewerBookmarks.tsx

```diff
@@ -47,6 +48,11 @@ type PreviouslyViewedLinksProps = {
 export const PreviouslyViewedLinks = (props: PreviouslyViewedLinksProps) => {
   const dispatch = useDispatch();
 
+  const onClick = () => {
+    dispatch(clearFilterSorting());
```

Comment:
The way I am reading the ticket, including the comments, every change to the active gene should reset the filters and sorting settings associated with it. This suggests that the proper place for clearing filters and sorting is within a redux action that sets the active gene rather than in any of the UI elements.

Reply:
What @azangru said makes sense and we also spotted a scenario where the filtering was being applied where it shouldn't. After a pair programming session, we have updated the code to make the filter state reset/clear in the EntityViewerGeneralSlice where the genomeId and entityId change is happening.